### PR TITLE
sync-single-account.py

### PR DESCRIPTION
**Pull request by squeaky-pl**
This is something that I did over and over using sync-engine console to debug escalations and otherwise troubleshoot syncing of a single account. Since it is not obvious how to do it I wrapped it in a script so everybody can use it.

### CODE REVIEW

File: inbox/mailsync/backends/base.py [hunk 1]
```diff
@@ -22,6 +26,8 @@ class MailsyncDone(InterruptibleThreadExit):
 
 
 class BaseMailSyncMonitor(InterruptibleThread):
+    sync_engine_class: ClassVar[type["FolderSyncEngine"]]
+
```
Comment (squeaky-pl) [inline inbox/mailsync/backends/base.py:30]: I've moved this out of an initializer so the thread class used to sync a single folder can be accessed without constructing a monitor. A monitor is a parent thread used to sync all the folders.

File: inbox/mailsync/backends/gmail.py [hunk 2]
```diff
@@ -55,174 +55,6 @@
 MAX_DOWNLOAD_COUNT = 1
 
 
-class GmailSyncMonitor(ImapSyncMonitor):
```
Comment (squeaky-pl) [inline inbox/mailsync/backends/gmail.py:58]: I just switched the order of classes here so I can access `GmailFolderSyncEngine` from `GmailSyncMonitor`

File: inbox/mailsync/service.py [hunk 3]
```diff
@@ -48,6 +50,24 @@ def shared_sync_event_queue_for_zone(zone):
...
+        module.PROVIDER: getattr(module, module.SYNC_MONITOR_CLS)
+        for module in module_registry.values()
+        if hasattr(module, "SYNC_MONITOR_CLS")
+    }
+
+    for provider_name, _ in providers.items():
+        if provider_name not in monitor_classes:
+            monitor_classes[provider_name] = monitor_classes["generic"]
+
+    return monitor_classes
```
Comment (squeaky-pl) [inline inbox/mailsync/service.py:68]: This was extracted from `SyncService` intializer so I can use this from the script. This code is itself terribly dynamic but sync-engine was built to be extended by installing external modules like plugins, something that we don't do ourselves but I don't want to change it right n ow.

File: inbox/mailsync/backends/gmail.py [hunk 4]
```diff
@@ -598,3 +430,172 @@ def g_msgids(namespace_id, session, in_):
         .all()
     )
     return {g_msgid for g_msgid, in query}
+
+
+class GmailSyncMonitor(ImapSyncMonitor):
+    sync_engine_class: ClassVar[type[FolderSyncEngine]] = GmailFolderSyncEngine
```
Comment (squeaky-pl) [inline inbox/mailsync/backends/gmail.py:436]: And this code was moved below with

`sync_engine_class: ClassVar[type[FolderSyncEngine]] = GmailFolderSyncEngine`

moved from the initializer to class body.

**Review by neob91-close — APPROVED** [review]

**Pull request merged**